src/openrct2/peep/Guest.cpp

```diff
@@ -6385,6 +6397,8 @@ static bool peep_find_ride_to_look_at(Peep* peep, uint8_t edge, uint8_t* rideToV
     tileElement = surfaceElement;
     do
     {
+        if (tileElement == nullptr)
+            break;
```

Comment:
How come you put this one inside the loop?

Reply:
Yeah true, that was my initial plan but then realized the existing check which did it right after calling map_get_surface_element as I wasn't entirely sure about the logic of this function given that it sometimes returns false and when the loop runs out of elements it would jump to the next loop, its quite confusing.